### PR TITLE
Add some goroutine leak checks

### PR DESCRIPTION
**Pull request by wulianglongrd**
Add some goroutine leak checks and fix some incorrect test cases.

### CODE REVIEW

File: pkg/kube/kclient/client_test.go [hunk 1]
```diff
@@ -280,6 +280,7 @@ func TestErrorHandler(t *testing.T) {
 	c.Kube().(*fake.Clientset).Fake.PrependReactor("*", "*", func(action k8stesting.Action) (bool, runtime.Object, error) {
 		return true, nil, fmt.Errorf("nope, out of luck")
 	})
+	c.RunAndWait(test.NewStop(t))
```
Comment (howardjohn) [inline pkg/kube/kclient/client_test.go:283]: why is this needed? seems like it should be valid to make a client without running?

Comment (wulianglongrd) [reply pkg/kube/kclient/client_test.go:283]: After the queue used internally by `NewFakeClient` is initialized, a new goroutine will be used to automatically start an infinite for loop, and a signal is required to stop it. The purpose here is to pass a stop signal so that it can exit, otherwise the checker says the goroutine is leaking.

https://github.com/kubernetes/client-go/blob/master/util/workqueue/delaying_queue.go#L141

L111 and L121
https://github.com/istio/istio/blob/master/pkg/kube/controllers/queue.go#L111

Comment (wulianglongrd) [reply pkg/kube/kclient/client_test.go:283]: > seems like it should be valid to make a client without running?

It looks like L280's ReactionFunc is working, rather than `NewFakeClient` not running.

File: pkg/kube/kclient/client_test.go [hunk 2]
```diff
@@ -288,6 +289,7 @@ func TestErrorHandler(t *testing.T) {
 func TestToOpts(t *testing.T) {
 	test.SetForTest(t, &features.InformerWatchNamespace, "istio-system")
 	c := kube.NewFakeClient()
+	c.RunAndWait(test.NewStop(t))
```
Comment (wulianglongrd) [inline pkg/kube/kclient/client_test.go:292]: The reason for this change is the same as [above](https://github.com/istio/istio/pull/50713/files#r1581734612).

**Review by hzxuzhonghu — APPROVED** [review]
